### PR TITLE
Enhance kernel version detection by using uname

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
* Refactored the detectKernelVersion function to not solely rely on vdsoVersion. Added a fallback mechanism using the KernelRelease function, which utilizes the uname system call to get the kernel release string.

---

I had `vDSO` disabled for tapping into some syscalls, so this is something good to have.

### CODE REVIEW

**Comment by brycekahle** [comment]
We moved away from `uname` because it is notoriously hard to parse, and often doesn't contain enough information to match the value the kernel expects (for kernels < 5.0).

You can always set `ProgramSpec.KernelVersion` if you don't want the library to automatically set the value.

**Comment by Vyom-Yadav** [comment]
> We moved away from uname because it is notoriously hard to parse, and often doesn't contain enough information to match the value the kernel expects (for kernels < 5.0).

@brycekahle I added this as a fallback mechanism. Setting the flag manually v/s at least having something to get the kernel version 🤷🏼‍♂️

**Comment by brycekahle** [comment]
I understand, but this is more code to maintain for what sounds like a very narrow use case (vDSO disabled). See the code we removed in #500 for the old parsing code.

Maybe the maintainers feel differently, but my vote is to not add more maintenance burden.

**Comment by Vyom-Yadav** [comment]
I can see why the team took that decision. I was using Tetragon, and it isn't working with vDSO disabled, so I was trying to work around that somehow. I don't program using eBPF so I don't know how to set this up using some external Tetragon config, if possible.

**Comment by Vyom-Yadav** [comment]
Indeed, it's not pretty, but that is just a backup option in case vdso is disabled.

It is not reducing any functionality, just enhancing it. It does not break anything. 

I have vDSO disabled to tap into time syscalls. This would help in debugging unreproducible builds, so that's the reason behind it.

**Comment by lmb** [comment]
Discussed this with @ti-mo, we're not going to bring back uname parsing. That said, I think we should allow running without the vDSO.

In fact, the current version of the library shouldn't really rely on the vDSO any more since https://github.com/cilium/ebpf/commit/fbff7dbd617fccbf0bea0ea8c383f162088c9df5 Can you check which version of the library you are using at the moment? My guess is that you don't have the commit above. If that isn't true, please open a bug report that includes the error you are seeing and which version of the library you are using. We might be able to work around this in some other fashion.

**Comment by Vyom-Yadav** [comment]
@lmb Created https://github.com/cilium/ebpf/issues/1408
